sklearn/decomposition/pca.py

```diff
@@ -494,9 +500,10 @@ def _fit_truncated(self, X, n_components, svd_solver):
         self.explained_variance_ratio_ = \
             self.explained_variance_ / total_var.sum()
         self.singular_values_ = S.copy()  # Store the singular values.
-        if self.n_components_ < n_features:
+        if self.n_components_ < min(n_features, n_samples):
```

Comment:
Could you add a test to make sure that `noise_variance_` is 0 when either `n_components > n_features` or `n_components > n_samples`?